snapcraft/internal/build_providers/_base_provider.py

```diff
@@ -87,8 +118,84 @@ def launch_instance(self) -> None:
             self.instance_name))
         self._launch()
 
+    def _disable_and_wait_for_refreshes(self):
+        # Disable autorefresh for 15 minutes,
+        # https://github.com/snapcore/snapd/pull/5436/files
+        now_plus_15 = datetime.datetime.now() + datetime.timedelta(minutes=15)
+        self._run(['sudo', 'snap', 'set', 'core', 'refresh.hold={}Z'.format(
+            now_plus_15.isoformat())])
+        # Auto refresh may have kicked in while setting the hold.
+        self.echoer.info('Waiting for pending snap auto refreshes.')
+        with contextlib.suppress(errors.ProviderExecError):
+            self._run(['sudo', 'snap', 'watch', '--last=auto-refresh'])
```

Comment:
Any chance we could hide the stderr, here? snapd prints an "error: no auto-refresh changes" or something similar when there are no changes that match the filter (which is often). That could throw folks off.

Reply:
error: no changes of type "auto-refresh" found

I don't have  a way to swallow output with mutlipass, but could be done with qemu. Can we add a bug for it instead?

Reply:
or maybe let's just get this fixed on the snapd side?

Reply:
I logged https://bugs.launchpad.net/snapd/+bug/1779768, not let's wait for @chipaca to take action :smile: